notificationdemo/source/objc/NotificationDemo/ViewController.m

```diff
@@ -56,65 +60,76 @@ - (void)viewDidLoad {
+            break;
+            
+        case ALERT_TYPE_CodeGreen:
+            [self addLogWithText:[NSString stringWithFormat:@"Notification body: %@", notification.alertMessage] textColor:[UIColor greenColor]];
+            break;
+            
+        case ALERT_TYPE_CodeYellow:
+            [self addLogWithText:[NSString stringWithFormat:@"Notification body: %@", notification.alertMessage] textColor:[UIColor yellowColor]];
+            break;
+            
+        default:
+            break;
+    }
 }
 
 - (void)onStarted {
-    [self addLogWithText:@"Kaa client started"];
+    [self addLogWithText:@"Kaa client started" textColor:nil];
 }
 
 - (void)onStopped {
-    [self addLogWithText:@"Kaa client stopped"];
+    [self addLogWithText:@"Kaa client stopped" textColor:nil];
 }
 
 - (void)onStartFailureWithException:(NSException *)exception {
-    [self addLogWithText:[NSString stringWithFormat:@"START FAILURE: %@ : %@", exception.name, exception.reason]];
+    [self addLogWithText:[NSString stringWithFormat:@"START FAILURE: %@ : %@", exception.name, exception.reason] textColor:nil];
 }
 
 - (void)onPaused {
-    [self addLogWithText:@"Client paused"];
+    [self addLogWithText:@"Client paused" textColor:nil];
 }
 
 - (void)onPauseFailureWithException:(NSException *)exception {
-    [self addLogWithText:[NSString stringWithFormat:@"PAUSE FAILURE: %@ : %@", exception.name, exception.reason]];
+    [self addLogWithText:[NSString stringWithFormat:@"PAUSE FAILURE: %@ : %@", exception.name, exception.reason] textColor:nil];
 }
 
 - (void)onResume {
-    [self addLogWithText:@"Client resumed"];
+    [self addLogWithText:@"Client resumed" textColor:nil];
 }
 
 - (void)onResumeFailureWithException:(NSException *)exception {
-    [self addLogWithText:[NSString stringWithFormat:@"RESUME FAILURE: %@ : %@", exception.name, exception.reason]];
+    [self addLogWithText:[NSString stringWithFormat:@"RESUME FAILURE: %@ : %@", exception.name, exception.reason] textColor:nil];
 }
 
 - (void)onStopFailureWithException:(NSException *)exception {
-    [self addLogWithText:[NSString stringWithFormat:@"STOP FAILURE: %@ : %@", exception.name, exception.reason]];
+    [self addLogWithText:[NSString stringWithFormat:@"STOP FAILURE: %@ : %@", exception.name, exception.reason] textColor:nil];
+}
+
+#pragma mark - UITextFieldDelegate
+
+- (BOOL)textFieldShouldReturn:(UITextField *)textField {
+    @try {
+        [self.kaaClient subscribeToTopicWithId:self.topicIdTextField.text.intValue];
+        [self addLogWithText:[NSString stringWithFormat:@"Subscribed to optional topic with id %d", self.topicIdTextField.text.intValue] textColor:nil];
+    } @catch (NSException *exception) {
+        [self addLogWithText:[NSString stringWithFormat:@"Topic with id %d is unavaliable. Can't subscribe", self.topicIdTextField.text.intValue] textColor:nil];
+    }
+    [textField resignFirstResponder];
+    return YES;
 }
 
 #pragma mark - Supporting methods
```

Comment:
Replace pragma block name as **ProfileContainer** for **getProfile** method (obviously, it was declared into ProfileContainer protocol)